packages/google-pay-integration/src/google-pay-payment-processor.ts

```diff
@@ -0,0 +1,238 @@
+    }
+
+    async setExternalCheckoutXhr(
+        provider: string,
+        response: GooglePayCardDataResponse,
+    ): Promise<void> {
+        return this._setExternalCheckout(provider, response);
+    }
+
+    async setExternalCheckoutForm(
+        provider: string,
+        response: GooglePayCardDataResponse,
+        siteLink?: string,
+    ): Promise<void> {
+        return this._setExternalCheckout(provider, response, true, siteLink);
+    }
+
+    mapToBillingAddressRequestBody(
+        response: GooglePayCardDataResponse,
+    ): BillingAddressRequestBody | undefined {
+        return this._gateway.mapToBillingAddressRequestBody(response);
+    }
+
+    mapToShippingAddressRequestBody(
+        response: GooglePayCardDataResponse,
+    ): AddressRequestBody | undefined {
+        return this._gateway.mapToShippingAddressRequestBody(response);
+    }
+
+    processAdditionalAction(error: unknown): Promise<void> {
+        return isGooglePayAdditionalActionProcessable(this._gateway)
+            ? this._gateway.processAdditionalAction(error)
+            : Promise.reject(error);
+    }
+
+    async signOut(providerId: string): Promise<void> {
+        await this._requestSender.get(`/remote-checkout/${providerId}/signout`);
+    }
+
+    async _setExternalCheckout(
+        provider: string,
+        response: GooglePayCardDataResponse,
+        useFormPoster = false,
+        siteLink?: string,
+    ): Promise<void> {
+        const url = '/checkout.php';
+        const body = {
+            action: 'set_external_checkout',
+            provider,
+            ...(await this._gateway.mapToExternalCheckoutData(response)),
+        };
+
+        if (useFormPoster) {
+            return new Promise((resolve) => {
+                this._formPoster.postForm(
+                    siteLink ? `${siteLink}/checkout` : url,
+                    {
+                        ...body,
+                        card_information: JSON.stringify(body.card_information),
+                    },
+                    resolve,
+                );
+            });
+        }
+
+        await this._requestSender.post(url, {
+            headers: {
+                Accept: 'text/html',
+                'Content-Type': 'application/x-www-form-urlencoded',
+                ...SDK_VERSION_HEADERS,
+            },
+            body,
+        });
+    }
+
+    private _prefetchGooglePaymentData(): void {
+        const paymentDataRequest = this._getPaymentDataRequest();
+
+        paymentDataRequest.transactionInfo = this._gateway.getTransactionInfo();
+
+        this._getPaymentsClient().prefetchPaymentData(paymentDataRequest);
+    }
+
+    private async _determineReadinessToPay(): Promise<void> {
+        try {
+            const { result } = await this._getPaymentsClient().isReadyToPay(
+                this._getIsReadyToPayRequest(),
+            );
+
+            if (result === false) {
+                throw new PaymentMethodFailedError(
```

Comment:
Hi @bc-nick. I can replicate your exact same fix on this PR but, since your change only affects customer strategy, I think it would be better to address it on my sibling PR #1994 in which I can catch the exception to avoid showing an error to the shopper if GPay is not supported. What do you think?